R/build-articles.R

```diff
@@ -374,3 +374,41 @@ default_articles_index <- function(pkg = ".") {
   ))
 
 }
+
+navbar_articles <- function(pkg = ".") {
+  pkg <- as_pkgdown(pkg)
+  vignettes <- pkg$vignettes
+
+  # if an explicit navbar is supplied in meta, use that; otherwise
+  # use a default-constructed navbar entry
+  meta_navbar <- pkg$meta$articles$navbar
+  if (is.null(meta_navbar))
+    return(menu("Articles", menu_links(vignettes$title, vignettes$file_out)))
+
+  # construct navbar (detect whether user has only provided menu entries,
+  # versus specific text + menu entries)
+  navbar <- meta_navbar
```

Comment:
Should we validate that the constructed navbar has the names `text` + `menu`? Only those names? Or at least those names?